From 721163458c3517da6996e754713a9b0800c720aa Mon Sep 17 00:00:00 2001
From: "Burla, Satananda" <sburla@marvell.com>
Date: Mon, 5 Oct 2020 09:16:25 -0700
Subject: [PATCH 08/19] octeontx2-dpi: Fix DPI engine blks allocation

commit b2dc7e46f346d65b32b02623f7db5cd6adfcdd57 from
git@git.assembla.com:cavium/WindRiver.linux.git

Octeontx2 support 64KB fifo, increase allocation

Signed-off-by: Burla, Satananda <sburla@marvell.com>
Change-Id: Idc675af2c182f2d6c5bef325ad450ff8a7193cf2
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/37348
Reviewed-by: Radha Chintakuntla <radhac@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Abhijit Ayarekar <aayarekar@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49031
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/soc/marvell/octeontx2-dpi/dpi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-dpi/dpi.c b/drivers/soc/marvell/octeontx2-dpi/dpi.c
index 0cdeb43a26f0..db5f84730888 100644
--- a/drivers/soc/marvell/octeontx2-dpi/dpi.c
+++ b/drivers/soc/marvell/octeontx2-dpi/dpi.c
@@ -140,9 +140,9 @@ static int dpi_init(struct dpipf *dpi)
 
 	for (engine = 0; engine < dpi_dma_engine_get_num(); engine++) {
 		if (engine == 4 || engine == 5)
-			reg = DPI_ENG_BUF_BLKS(8);
+			reg = DPI_ENG_BUF_BLKS(16);
 		else
-			reg = DPI_ENG_BUF_BLKS(4);
+			reg = DPI_ENG_BUF_BLKS(8);
 
 		dpi_reg_write(dpi, DPI_ENGX_BUF(engine), reg);
 
-- 
2.31.1

